sklearn/linear_model/tests/test_ransac.py

```diff
@@ -494,3 +494,17 @@ def test_ransac_fit_sample_weight():
     base_estimator = Lasso()
     ransac_estimator = RANSACRegressor(base_estimator)
     assert_raises(ValueError, ransac_estimator.fit, X, y, weights)
+
+
+def test_ransac_base_estimator_fit_sample_weight():
+    class DummyLinearRegression(LinearRegression):
+        def fit(self, X, y, sample_weight):
+            super().fit(X, y, sample_weight)
+
+    base_estimator = DummyLinearRegression()
+    ransac_estimator = RANSACRegressor(base_estimator, random_state=0)
+    n_samples = y.shape[0]
+    weights = np.ones(n_samples)
```

Comment:
I don't think that the test is actually testing anything. We should make sure that passing non unit weights will lead to a final model train on non unit weights. One way to do that is to pass a `sample_weight` with non unit weights and be sure sure that some of these weights will be used. Then we can use `ransac_estimator.inlier_mask_` to train a model which should give the same results than the fitted model in ransac.